### PR TITLE
feat: Grain implementation of equals

### PR DESCRIPTION
**Pull request by ospencer**

### CODE REVIEW

File: compiler/src/codegen/compcore.re [hunk 1]
```diff
@@ -251,9 +261,17 @@ let grain_function_imports = [
...
   {
-    mimp_mod: stdlib_external_runtime_mod,
+    mimp_mod: equal_mod,
     mimp_name: equal_ident,
-    mimp_type: MFuncImport([I32Type, I32Type], [I32Type]),
+    mimp_type: MFuncImport([I32Type, I32Type, I32Type], [I32Type]),
+    mimp_kind: MImportWasm,
+    mimp_setup: MSetupNone,
+  },
+  // HACK: Depend on stdlib-external/runtime for the tests
```
Comment (ospencer) [inline compiler/src/codegen/compcore.re:270]: I'm actually not really sure what we're going to do about this—the only thing that always depends on `toString` is `Pervasives`, and programs that don't depend on `Pervasives` won't have loaded this module. This hack also only works until #196 lands.

I think I'd probably want a lightweight version of this implemented solely in JS, and any test that wants to test printing enums/records has to call `print` explicitly.

There's also a solid chance that we'll need to change how `_start` works to not return a value at all, which makes this a bit of a moot point as most tests will need to be rewritten to use `print` or some other technique.

Comment (phated) [reply compiler/src/codegen/compcore.re:270]: We're getting optimizations now, right? So what are we going to do?

File: stdlib/pervasives.gr [hunk 2]
```diff
@@ -75,7 +77,7 @@ primitive fail : String -> a = "@fail"
 import foreign wasm print : a -> Void from "stdlib-external/runtime"
 
 // Checks the given items for structural equality.
-import foreign wasm equal : (a, a) -> Bool as (==) from "stdlib-external/runtime"
+export (==)
```
Comment (phated) [inline stdlib/pervasives.gr:80]: We already `export *` at top?

File: stdlib/runtime/equal.gr [hunk 3]
```diff
@@ -0,0 +1,209 @@
+/* compilation-mode: runtime */
```
Comment (phated) [inline stdlib/runtime/equal.gr:1]: Mostly taking your word for this conversion. I don't have the energy to try to line up the AssemblyScript side-by-side with the Grain code.

**Review by phated — APPROVED** [review]

File: compiler/src/codegen/compcore.re (hunk 1 above)
Comment (ospencer) [reply compiler/src/codegen/compcore.re:270]: I actually managed to resolve this in #540.

File: stdlib/pervasives.gr (hunk 2 above)
Comment (ospencer) [reply stdlib/pervasives.gr:80]: Doesn't work on imports.

**Review by peblair — APPROVED** [review]

**Pull request merged**